src/test/scala/uk/gov/hmrc/play/frontend/bootstrap/FilterChainExceptionSpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.play.frontend.bootstrap
+
+import org.scalatest.{Matchers, WordSpecLike}
+import play.api.libs.concurrent.Execution.{defaultContext => ec}
+import play.api.mvc._
+import play.api.test.{FakeApplication, WithServer, WsTestClient}
+import play.filters.headers.SecurityHeadersFilter
+import uk.gov.hmrc.play.filters.RecoveryFilter
+import uk.gov.hmrc.play.http.NotFoundException
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class FilterChainExceptionSpec extends WordSpecLike with Matchers with WsTestClient {
+
+  val routerForTest: PartialFunction[(String, String), Handler] = {
+    case ("GET", "/ok") => Action { request => Results.Ok("OK") }
```

Comment:
Please make the above as close to the 404 case as possible - i.e. return a Future of the Result.